Code/GraphMol/AddHs.cpp

```diff
@@ -326,35 +325,66 @@ void setHydrogenCoords(ROMol *mol, unsigned int hydIdx, unsigned int heavyIdx) {
           }
         } else {
           // we're in flatland
-          // this was github #908
-          // We're in a 2D conformation, put the H between the two neighbors
-          // that have the widest angle between them. Unless the two are
-          // opposite ends of a straight line through the heavy atom,
-          // which would make the H overlap with the heavy atom.
-          // In such case, set it on the opposite direction to the 3rd neighbor.
-          double minDot = nbr1Vect.dotProduct(nbr2Vect);
-          if (fabs(minDot + 1) < 1e-4) {
-            dirVect = -nbr3Vect;
-          } else {
-            dirVect = nbr1Vect + nbr2Vect;
+          double dot12 = nbr1Vect.dotProduct(nbr2Vect);
+          double dot13 = nbr1Vect.dotProduct(nbr3Vect);
+          double dot23 = nbr2Vect.dotProduct(nbr3Vect);
+          unsigned int numNeg = 0;
+          if (dot12 < -1e-4) {
+            ++numNeg;
           }
-          if (nbr2Vect.dotProduct(nbr3Vect) < minDot) {
-            minDot = nbr2Vect.dotProduct(nbr3Vect);
-            if (fabs(minDot + 1) < 1e-4) {
-              dirVect = -nbr1Vect;
+          if (dot13 < -1e-4) {
+            ++numNeg;
+          }
+          if (dot23 < -1e-4) {
+            ++numNeg;
+          }
+
+          if (1) {  // numNeg < 3) {
```

Comment:
Did this slip in? Because if not, then the other branch is redundant and should be removed.

Still, I made some quick sketches, and this approach seems to make some sense, although we'd end up with some more symmetry, since we basically just place the new H atom on the same vector as one of the existing neighbors, just at the opposite side of the central atom.

Reply:
Apologies; that should have been cleaned up. I pushed the wrong version.

After reviewing the IUPAC guidelines, I'm going to revise this and place the H so that it bisects the largest angle instead of being opposite the third bond. Here's the relevant excerpt from the IUPAC doc:
![image](https://user-images.githubusercontent.com/540511/110422252-4b838d00-809f-11eb-91bf-bf4f674703cf.png)

Reply:
> Apologies; that should have been cleaned up. I pushed the wrong version.

That what reviews are for! :D